packages/data-stores/src/site/reducer.ts

```diff
@@ -38,25 +43,40 @@ const newSiteError: Reducer< NewSiteErrorResponse | undefined, Action > = ( stat
 	return state;
 };
 
-const isFetchingSite: Reducer< boolean | undefined, Action > = ( state = false, action ) => {
+export const isFetchingSite: Reducer< boolean | undefined, Action > = ( state = false, action ) => {
 	switch ( action.type ) {
 		case 'FETCH_NEW_SITE':
 			return true;
 		case 'RECEIVE_NEW_SITE':
-			return false;
 		case 'RECEIVE_NEW_SITE_FAILED':
+		case 'RESET_SITE_STORE':
 			return false;
 	}
 	return state;
 };
 
+export const existingSite: Reducer<
```

Comment:
How about a shared `sites` reducer for both newly created, and existing sites? (Again inspired by Calypso.) My mental model here is that the new/existing distinction is something quite ephemeral -- a new site becomes 'existing' rather quickly.

Reply:
(To clarify, Calypso's `sites` reducer contains all of a user's sites, i.e. basically the response from `/me/sites`. However, it's perfectly conceivable to add sites to it only piecemeal.)

Reply:
Unfortunately I think we might still need to keep the newly created site data from the existing one, because the response object returned from `sites/new` is of quite a different shape to the one returned by `/sites`. For example in a newly created site, the response object is quite minimal:

```
{
  "url": "https://andysfakesite284705224.wordpress.com/",
  "blogid": "174855534",
  "blogname": "Andy's fake site",
  "xmlrpc": "https://andysfakesite284705224.wordpress.com/xmlrpc.php",
  "site_slug": "andysfakesite284705224.wordpress.com"
}
```

Whereas the same site after calling the `/sites` endpoint will be the full site object. I think it's a great idea to include a newly created site in the `sites` reducer, but to do that, I think we'd have to get the consumer to call `getSite` or we could fire off an API request from this data store to do that additional call (triggering the resolver for example)... but for the purposes of site creation we mightn't always need that extra request. I've left these separate for the moment, I think it might make sense to burrow further into this in another PR now that this one has gotten a bit complex, unless you can think of a better way to handle it?

Reply:
Yeah, we can defer that. Not always needing the full site information upon site creation is a very good argument.

We _could_ consider marking the extra information that the `/sites` endpoint gives us as optional (mostly talking about TS defs here), but that sort of heterogenity in the state tree might prove problematic. Let's punt this for now.